src/shared/install.c

```diff
@@ -1056,11 +1058,13 @@ static int config_parse_also(
                 if (r == 0)
                         break;
 
-                r = install_info_add(c, word, NULL, NULL);
+                r = install_info_add(c, word, NULL, &alsoinfo);
                 if (r < 0)
                         return r;
+                if (r == 1) /* new UnitFileInstallInfo */
+                        alsoinfo->auxiliary = true;
```

Comment:
I was trying to cover the case, that's why I added the return value to add_install_info and look at it. But my logic is borked: it only works if B would be specified before A on the command-line, and not the other way around. Will fix.